projects/parachain-demo/src/index.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+// hack to make poladot-js work without bringing in webpack and babel
+import "regenerator-runtime/runtime"
+import { Detector }  from '@substrate/connect';
+import westmint from './assets/westend-westmint.json';
+import UI, { emojis } from './view';
+
+window.onload = () => {
+  const loadTime = performance.now();
+  const ui = new UI({ containerId: 'messages' }, { loadTime });
+  ui.showSyncing();
+
+  (async () => {
+    try {
+      const detect = new Detector('Parachain Demo');
+      const api = await detect.connect('westend2', JSON.stringify(westmint));
+      
+      const [chain, nodeName, nodeVersion, properties] = await Promise.all([
+        api.rpc.system.chain(),
+        api.rpc.system.name(),
+        api.rpc.system.version(),
+        api.rpc.system.properties()
+      ]);
+      const header = await api.rpc.chain.getHeader()
+      const chainName = await api.rpc.system.chain()
+
+      // Show chain constants - from chain spec
+      ui.log(`${emojis.seedling} Light client ready - Using ${chain} - ${nodeName}: ${nodeVersion}`, true);
+      ui.log(`${emojis.info} Connected to ${chainName}: syncing will start at block #${header.number}`);
+      ui.log(`${emojis.chequeredFlag} Token decimals: ${properties['tokenDecimals']} - symbol: ${properties['tokenSymbol']}`);
+      ui.log(`${emojis.chequeredFlag} Genesis hash is ${api.genesisHash.toHex()}`);
+
+      // Show how many peers we are syncing with
+      const health = await api.rpc.system.health();
+      const peers = health.peers.toNumber() === 1 ? '1 peer' : `${health.peers} peers`;
+      ui.log(`${emojis.stethoscope} Chain is syncing with ${peers}`);
+
+      const waitForChainToSync = () => {
+        let resolved = false;
+        return new Promise<void>((resolve, reject) => {
+          setInterval(() => {
+            api.rpc.system.health().then(health => {
+              if (health.isSyncing.eq(false)) {
+                ui.showSynced();
+                if (!resolved) {
+                  resolved = true;
+                  resolve();
+                }
+              } else {
+                ui.showSyncing();
+              }
+            }).catch(error => {
+              ui.error(error);
+              if (!resolved) {
+                resolved = true;
+                reject();
+              }
+            });
+          }, 2000);
+        });
+      }
```

Comment:
The `setInterval` is never cleared and the error is not propagated through `reject`.

The following implementation addresses both things:

```suggestion
      const wait = (ms: number) => new Promise<void>(res => { setTimeout(res, ms); })

      const waitForChainToSync = async () => {
        const health = await api.rpc.system.health();

        if (health.isSyncing.eq(false)) {
          ui.showSynced();
        } else {
          ui.showSyncing();
          await wait(2000);
          await waitForChainToSync();
        }
      }
```

Also, in case someone is wondering if this could cause a stackoverflow: it won't because the `wait` function creates a macrotask, which "resets" the event-loop.